db-service/lib/fill-in-keys.js

```diff
@@ -57,9 +57,12 @@ const generateUUIDandPropagateKeys = (entity, data, event) => {
 module.exports = async function fill_in_keys(req, next) {
   // REVISIT dummy handler until we have input processing
   if (!req.target || !this.model || req.target._unresolved) return next()
-  if (req.event === 'UPDATE') {
+  // only for deep update
+  const containsObjectOrArray = obj =>
```

Comment:
This check already exists inside the `deep-queries.js`. Instead make sure this function is only called when it should be called. If it is not required for any queries where it does not deep then it should be made sure this function is not called.

As this check might identify a value which is actually ignored. Like an association.

Reply:
would change this to: `req.event === 'UPDATE' && hasDeep(req.query, req.target)`